statsmodels/graphics/_probscale.py

```diff
@@ -0,0 +1,177 @@
+import numpy as np
+import matplotlib
+from matplotlib.transforms import Transform
+from matplotlib.scale import ScaleBase
+from matplotlib.ticker import (
+    FixedLocator,
+    NullLocator,
+    Formatter,
+    NullFormatter,
+    FuncFormatter
+)
+import matplotlib.pyplot as plt
+from scipy import stats
+
+
+def _get_probs(nobs):
+    '''Returns the x-axis labels for a probability plot based
+    on the number of observations (`nobs`)
+    '''
+    order = int(np.floor(np.log10(nobs)))
+    base_probs = np.array([10., 20., 30., 40., 50., 60., 70., 80., 90.])
+
+    axis_probs = base_probs.copy()
+    for n in range(order):
+        if n <= 2:
+            lower_fringe = np.array([1., 2., 5.])
+            upper_fringe = np.array([5., 8., 9.])
+        else:
+            lower_fringe = np.array([1.])
+            upper_fringe = np.array([9.])
+
+        new_lower = lower_fringe/10.**(n)
+        new_upper = upper_fringe/10.**(n) + axis_probs.max()
+        axis_probs = np.hstack([new_lower, axis_probs, new_upper])
+
+    return axis_probs
+
+
+def _sig_figs(val, nsf, expthresh=5):
+    '''Formats a number into a string with the correct number of sig figs.
+
+    Parameters
+    -------s---
+    val : numeric
+        the number you want to round
+    nsf : int
+        the number of sig figs it should have
+
+    Example
+    -------
+    >>> print(_sig_figs(1247.15, 3))
+    1250
+    >>> print(_sig_figs(1247.15, 7))
+    1247.150
+
+    '''
+    # check on the number provided
+    if not np.isfinite(val):
+        raise ValueError("`val` must be a finite number")
+
+    else:
+
+        # check on the _sig_figs
+        if nsf < 1:
+            raise ValueError("number of sig figs must be greater than zero!")
+
+        # logic to do all of the rounding
+        elif val != 0.0:
+            order = np.floor(np.log10(np.abs(val)))
+
+            if -1.0 * expthresh <= order <= expthresh:
+                decimal_places = int(nsf - 1 - order)
+
+                if decimal_places <= 0:
+                    out = '{0:,.0f}'.format(round(val, decimal_places))
+
+                else:
+                    fmt = '{0:,.%df}' % decimal_places
+                    out = fmt.format(val)
+
+            else:
+                decimal_places = nsf - 1
+                fmt = '{0:.%de}' % decimal_places
+                out = fmt.format(val)
+
+        else:
+            out = str(round(val, nsf))
+
+    return out
+
+
+class ProbFormatter(Formatter):
```

Comment:
all classes and methods should have at least minimal docstring, even if internal.